src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  const dateParts = date.split(fromSeparator);
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dateParts[i];
+    } else if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    } else if (fromFormat[i] === 'DD') {
+      day = dateParts[i];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && fromFormat.includes('YY')) {
+    year = year < 30 ? `20${year}` : `19${year}`;
+  } else if (toFormat.includes('YY') && fromFormat.includes('YYYY')) {
+    year = year.slice(-2);
+  }
+
+  const newDateParts = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'YYYY' || toFormat[i] === 'YY') {
+      newDateParts.push(year);
+    } else if (toFormat[i] === 'MM') {
+      newDateParts.push(month);
+    } else if (toFormat[i] === 'DD') {
+      newDateParts.push(day);
+    }
+  }
+
+  return newDateParts.join(toSeparator);
```

Comment:
The loop for constructing the new date parts assumes that the format arrays do not include the separator as part of the format strings. Ensure that this assumption holds true for all input cases or adjust the logic to handle separators correctly.